kpops/pipeline_generator/pipeline.py

```diff
@@ -137,13 +138,21 @@ def load_from_yaml(
         :returns: Initialized pipeline object
         """
         Pipeline.set_pipeline_name_env_vars(base_dir, path)
+        Pipeline.set_environment_name(environment)
 
         main_content = load_yaml_file(path, substitution=ENV)
         if not isinstance(main_content, list):
             msg = f"The pipeline definition {path} should contain a list of components"
             raise TypeError(msg)
         env_content = []
-        if (env_file := Pipeline.pipeline_filename_environment(path, config)).exists():
+        if (
+            ENV.get("environment") is not None
```

Comment:
You can only do this:
```suggestion
            environment is not None
```
Or is there a reason you read the ENV dictionary again?